### PR TITLE
4 to main

### PR DESCRIPTION
**Pull request by adlarkin**
Forward port `ign-gazebo4` to `main`.

### CODE REVIEW

**Review by chapulina — COMMENTED** [review]
Besides the `pre1` comment, this PR also needs to wait for https://github.com/ignitionrobotics/ign-rendering/pull/178 to go in, and we'll need to bump the `ign-rendering5` commit here for Homebrew CI to pass:

https://github.com/osrf/homebrew-simulation/blob/35f76dfe47a67ed1e4d52064724165cb9df35ee6/Formula/ignition-rendering5.rb#L4-L6

File: CMakeLists.txt
Comment (adlarkin) [reply CMakeLists.txt]: Added back in: https://github.com/ignitionrobotics/ign-gazebo/pull/460/commits/60c793649cb64dd4d04200ed9eeef36922ac39cc

**Comment by adlarkin** [comment]
> Besides the `pre1` comment, this PR also needs to wait for [ignitionrobotics/ign-rendering#178](https://github.com/ignitionrobotics/ign-rendering/pull/178) to go in, and we'll need to bump the `ign-rendering5` commit here for Homebrew CI to pass:
> 
> https://github.com/osrf/homebrew-simulation/blob/35f76dfe47a67ed1e4d52064724165cb9df35ee6/Formula/ignition-rendering5.rb#L4-L6

Looks like https://github.com/ignitionrobotics/ign-rendering/pull/178 is ready to be merged. Regarding the `ign-rendering5` bump for Homebrew, does that just mean we need to update the [sha256 line](https://github.com/osrf/homebrew-simulation/blob/35f76dfe47a67ed1e4d52064724165cb9df35ee6/Formula/ignition-rendering5.rb#L6) to match the commit hash that corresponds to the merge of https://github.com/ignitionrobotics/ign-rendering/pull/178?

**Comment by chapulina** [comment]
> Looks like ignitionrobotics/ign-rendering#178 is ready to be merged.

Yup, approved. Note that after that's merged, we'll have to wait one night for a new nightly release of `ign-rendering5` that's used by the Actions CI.

>  we need to update the sha256 line to match the commit hash that corresponds to the merge of ignitionrobotics/ign-rendering#178?

Exactly, here's an example PR: https://github.com/osrf/homebrew-simulation/pull/1086/. Note how the `url`, `version` and `sha256` fields are updated.

**Comment by adlarkin** [comment]
The connecting PR that updates `ign-rendering5` is now up: https://github.com/osrf/homebrew-simulation/pull/1214

**Comment by adlarkin** [comment]
https://github.com/osrf/homebrew-simulation/pull/1214 has been merged, so nothing should be blocking this PR anymore - all that's needed is a review.

**Comment by chapulina** [comment]
@osrf-jenkins run tests again? Thanks!

**Comment by adlarkin** [comment]
It looks like Homebrew CI still failed. I tried looking into the failure, but when I click on the link, I keep getting faced with a `502 Bad Gateway` error message. Is anyone else able to view why Homebrew CI failed? I think Jenkins might be down...

Also, it looks like GitHub Ubuntu CI had the following failures, while Jenkins Ubuntu had no failures (I'm not sure why they'd differ - different testing hardware, perhaps?):

```
  97% tests passed, 4 tests failed out of 136
  
  Total Test time (real) = 843.61 sec
  
  The following tests FAILED:
  	 23 - UNIT_Server_TEST (Timeout)
  	 37 - UNIT_PeerTracker_TEST (Failed)
  	125 - INTEGRATION_log_system (Failed)
  	133 - PERFORMANCE_each (Failed)
```

**Review by chapulina — APPROVED** [review]
> Is anyone else able to view why Homebrew CI failed?

It looks like it's back up now. There's one flaky test failure, `UNIT_PeerTracker_TEST`, which hopefully will be fixed when #452 is merged forward.

> GitHub Ubuntu CI had the following failures, while Jenkins Ubuntu had no failures (I'm not sure why they'd differ - different testing hardware, perhaps?):

Yes, the setup and hardware are a little different and allow us to catch different problems. Actions is usually less more sensitive about timing. From memory, I believe none of those are new failures. It's tricky because we don't have a good way to track test failures over time with actions. Another reason to fix them asap, so we don't need to remember which ones are broken :smile:

**Pull request merged**
